viewer/packages/cad-geometry-loaders/src/sector/culling/takensectors/TakenSectorMapBase.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+import { PrioritizedWantedSector, SectorCost, SectorLoadingSpent } from '../types';
+import { CadModelMetadata, LevelOfDetail } from '@reveal/cad-parsers';
+
+export abstract class TakenSectorMapBase {
+  protected abstract get modelsMetadata(): CadModelMetadata[];
+  abstract get totalCost(): SectorCost;
+  abstract collectWantedSectors(): PrioritizedWantedSector[];
+
+  computeSpentBudget(): SectorLoadingSpent {
+    const modelsMetadata = this.modelsMetadata;
+    const wanted = this.collectWantedSectors();
+    const nonDiscarded = wanted.filter(sector => sector.levelOfDetail !== LevelOfDetail.Discarded);
+
+    const totalSectorCount = modelsMetadata.reduce((sum, x) => sum + x.scene.sectorCount, 0);
+    const takenSectorCount = nonDiscarded.length;
+    const takenSimpleCount = nonDiscarded.filter(x => x.levelOfDetail === LevelOfDetail.Simple).length;
+    const forcedDetailedSectorCount = nonDiscarded.filter(x => !Number.isFinite(x.priority)).length;
+    const accumulatedPriority = nonDiscarded
+      .filter(x => Number.isFinite(x.priority) && x.priority > 0)
+      .reduce((sum, x) => sum + x.priority, 0);
```

Comment:
I prefer when the lambda functions have descriptive variable names, but I'll let you make that choice.